app/services/omniauth_sign_in_service.rb

```diff
@@ -0,0 +1,9 @@
+class OmniauthSignInService
+  def find_or_create_user(access_token:)
+    data = access_token.info
+
+    User.find_or_create_by(email: data['email']) do |user|
```

Comment:
I think here you need to use `find_or_create_by!` with bang so to raise an exception if something goes wrong